### PR TITLE
Bundle external data into docker image, rather than download at runtime

### PR DESCRIPTION
**Pull request by pankajroark**
To benefit from faster startup times of docker images. Uses requests by default, but if b10cp path is supplied as env var then uses that.

### CODE REVIEW

File: truss/contexts/image_builder/serving_image_builder.py [hunk 1]
```diff
@@ -144,3 +157,82 @@ def _render_dockerfile(
...
+    return os.environ.get(B10CP_PATH_TRUSS_ENV_VAR_NAME)
+
+
+def _download_external_data_using_b10cp(
+    b10cp_path: str,
+    data_dir: Path,
+    external_data: ExternalData,
+):
+    procs = []
+    for item in external_data.items:
```
Comment (bolasim) [inline truss/contexts/image_builder/serving_image_builder.py:196]: Is there any downside to not limiting the concurrency here?

File: truss/contexts/image_builder/serving_image_builder.py [hunk 2]
```diff
@@ -144,3 +157,82 @@ def _render_dockerfile(
...
+        path = (data_dir / item.local_data_path).resolve()
+        proc = _download_from_url_using_b10cp(b10cp_path, item.url, path)
+        procs.append(proc)
+
+    for proc in procs:
+        proc.wait()
```
Comment (bolasim) [inline truss/contexts/image_builder/serving_image_builder.py:202]: should we check that these processes succeeded?

**Review by bolasim — APPROVED** [review]
lgtm. just a few questions.

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "truss"
-version = "0.4.8dev0"
+version = "0.4.9rc1"
```
[the remark this answers is not in the document]
Comment (pankajroark) [reply pyproject.toml:3]: Let me revert this.

File: truss/contexts/image_builder/serving_image_builder.py (hunk 1 above)
Comment (pankajroark) [reply truss/contexts/image_builder/serving_image_builder.py:196]: Good point. Not currently, we don't have external data used extensively right now. But let me leave a todo here.

File: truss/contexts/image_builder/serving_image_builder.py (hunk 2 above)
Comment (pankajroark) [reply truss/contexts/image_builder/serving_image_builder.py:202]: Good point, let me add a log here.

Comment (pankajroark) [reply truss/contexts/image_builder/serving_image_builder.py:203]: Let me do in a follow-up, to unblock for now.

**Pull request merged**